base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -432,13 +430,14 @@ private static ACMEOrder loadOrder(LDAPEntry entry)
         }
 
         List<ACMEIdentifier> identifiers = new ArrayList<>();
-        for (String type : IDENTIFIER_TYPES) {
-            attr = entry.getAttribute(ATTR_IDENTIFIER + ";" + type);
-            for (String value : Collections.list(attr.getStringValues())) {
-                identifiers.add(new ACMEIdentifier(type, value));
-            }
+        attr = entry.getAttribute(ATTR_IDENTIFIER);
+        for (String identifier : attr.getStringValueArray()) {
+            String[] parts = StringUtils.split(identifier, ":", 2);
+            String type = parts[0];
+            String value = parts[1];
+            identifiers.add(new ACMEIdentifier(type, value));
```

Comment:
Ad-hoc attribute subtypes via ';' is standard LDAP stuff; it's disappointing OpenLDAP apparently doesn't support it properly.  And this is a breaking change for existing databases.  So could we put this behavoiur in the `OpenLDAPDatabase` class only, and leave `LDAPDatabase` as-is?  (This comment also applies at several other place in the diff).

Reply:
I think we should minimize the difference between LDAP implementations to make it easier to migrate from one database to another. The ACME is a tech preview in PKI 10.8 and 10.9 and we've already [warned](https://github.com/dogtagpki/pki/blob/master/docs/installation/acme/Installing_PKI_ACME_Responder.md) about possible incompatible changes. Note that we don't have any automated database upgrade mechanism, so it's better to fix the schema before ACME becomes officially supported. Also, the ACME database is supposed to be transient anyway, so people can just remove their existing ACME subtree.

Reply:
After further research, I agree with changing it.  Although the LDAP data model allows arbitrary attribute description options (tags), they are supposed to be registered with IANA and an implementation SHALL treat an attribute description with an unrecognised attribute option as unrecognised.  ( https://tools.ietf.org/html/rfc4512#section-2.5 ).